code/__HELPERS/lists.dm

```diff
@@ -11,24 +11,38 @@
 
 //Returns a list in plain english as a string
 /proc/english_list(var/list/input, nothing_text = "nothing", and_text = " and ", comma_text = ", ", final_comma_text = "" )
-	var/total = input.len
-	if (!total)
+	var/list/types = uniquetypelist(input)
+	var/uniquetotal = types.len
+	var/typecount = 0
+	var/currentName = ""
+	if (!uniquetotal)
 		return "[nothing_text]"
-	else if (total == 1)
-		return "[input[1]]"
-	else if (total == 2)
-		return "[input[1]][and_text][input[2]]"
+	else if (uniquetotal == 1)
+		typecount = prune_list_to_type(input, types[1]).len
```

Comment:
This is a dangerous usage of Byond's permissive syntax and compilation rules. There is a reason why we have a check in our linter against this.
You need to put a list variable for the proc and then use the `.len` attribute of this var.
```dm
var/list/L = prune_list_to_type(input, types[1])
typecount = L.len
```
And so on.

Reply:
Done